src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,7 +1625,16 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
-                return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                if (!request.isSilent() && callbackHandle.callback != null && activity != null) {
+                    return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                } else {
+                    // cause exception added to indicate that refresh token was wiped
+                    callbackHandle.onError(new AuthenticationException(
```

Comment:
With the refactor I'm doing with another PR, i don't think this is necessary anymore.